tethys_gizmos/views/gizmos/jobs_table.py

```diff
@@ -56,36 +71,96 @@ def show_log(request, job_id):
         # Get the Job logs.
         data = job.get_logs()
 
-        success = True
-        message = ''
+        sub_job_options = [(k, k) for k in data.keys()]
+        sub_job_select = SelectInput(
+            display_text='Select Log:',
+            name='sub_job_select',
+            multiple=False,
+            options=sub_job_options,
+            attributes={'data-job-id': job_id}
+        )
+
+        context = {'sub_job_select': sub_job_select, 'log_options': {}}
+
+        for k, v in data.items():
+            if isinstance(v, dict):
+                context['log_options'][f'log_select_{k}'] = SelectInput(
+                    name=f'log_{k}',
+                    multiple=False,
+                    options=[(key, key) for key in v.keys()],
+                )
+
+        html = render_to_string('tethys_gizmos/gizmos/job_logs.html', context)
+
+        def remove_callables(item):
```

Comment:
Should remove_callable be handled  within the call to get_logs? So that get_logs return something fairly simple to parse without a lot of post processing that requires knowledge of how it is working underneath? Maybe I’m not understanding everything that is going on here.

Reply:
The `remove_callables` function handles some post processing so that it can be passed to JSON. In most cases when calling the `get_logs` method you want the callables to be there so that you can call them to get the log content.